edx_solutions_api_integration/users/views.py

```diff
@@ -1643,3 +1644,41 @@ def get(self, request, *args, **kwargs):  # pylint: disable=unused-argument
         })
 
         return Response(serializer.data, status=status.HTTP_200_OK)
+
+
+class UsersListWithEnrollment(UsersList):
+    """
+    View to create Users and enroll them in a list of courses.  In addition to
+    the options provided by UsersList.post, this view accepts an optional
+    "courses" attribute in the request body, which is a list of course keys to
+    enroll in.  
+    
+    The response will be annotated with a "courses" attribute which reflects
+    the list of courses the user was successfully enrolled in.  Failure to
+    enroll in a given course will not propagate an error to the caller, it will
+    just cause that course's key to be omitted from the response.
+    """
+
+    def post(self, request):
+        """
+        POST /api/users/integration_test_users/
+        """
+        AUDIT_LOG.warning(
+            "API::Creating and enrolling user with UsersListWithEnrollment. "
+            "This should not be used in production"
+        )
+        response = super(UsersListWithEnrollment, self).post(request)
+        if response.status_code == status.HTTP_201_CREATED:
+            user = User.objects.get(username=request.data['username'])
+            response.data['courses'] = []
+            course_key_gen = (CourseKey.from_string(course) for course in request.data['courses'])
```

Comment:
Could you please move the `CourseKey.from_string` call into the `try` block below, and catch `opaque_keys.InvalidKeyError` as well as `CourseEnrollmentException`?

It's not a huge deal, since we're only using this method for integration testing, but it's still nice to have.